### PR TITLE
Fix unwanted TailStrategy::Auto behavior in corner case

### PR DESCRIPTION
**Pull request by abadams**
@psuriana

### CODE REVIEW

File: src/Func.cpp [hunk 1]
```diff
@@ -966,19 +967,22 @@ void Stage::split(const string &old, const string &outer, const string &inner, E
...
+                auto it = descends_from_shiftinwards_outer.find(s.old_var);
                 if (s.is_split() && s.tail == TailStrategy::ShiftInwards) {
-                    descends_from_shiftinwards_outer.insert(s.outer);
-                } else if (s.is_split() && descends_from_shiftinwards_outer.count(s.old_var)) {
-                    descends_from_shiftinwards_outer.insert(s.inner);
-                    descends_from_shiftinwards_outer.insert(s.outer);
+                    descends_from_shiftinwards_outer[s.outer] = s.factor;
+                } else if (s.is_split() && it != descends_from_shiftinwards_outer.end()) {
+                    descends_from_shiftinwards_outer[s.inner] = it->second;
+                    descends_from_shiftinwards_outer[s.inner] = it->second;
```
Comment (psuriana) [inline src/Func.cpp]: Why is this duplicated?

Comment (abadams) [reply src/Func.cpp]: Oops, fixed.

**Comment by dsharletg** [comment]
The test failures look unrelated, should this be merged?

**Comment by abadams** [comment]
I'll update to master and retry the failing tests

**Comment by abadams** [comment]
Looks like llvm 5 on travis is busted.

**Pull request merged**